### DISCUSSION TITLE
Progress bar

### DISCUSSION THREAD

**Discussion by pehkawn** [amices/mice#551] [Ideas]
Imputing large datasets may often require a considerable amount of time, and the user have very little info on how the imputation process is progressing. Having some feedback on the progress, such as a progress bar, would've been nice.

In the documentation it states that `futuremice()` is based on `furrr::futuremap()`. During an unrelated search, I became aware that `furrr::futuremap()` comes with a `.progress` parameter. I tried passing `.progress = TRUE` in `futuremice()`, but the parameter is effectively ignored. Perhaps it would be possible to use `futuremap()`s progress bar to see the progress of the imputations?

**Comment by thomvolker** [comment]
Thanks for the suggestion!

I agree that this would be a very welcome addition. However, the `.progress` argument will be deprecated in `furrr`, which is why this is not implemented. An optional progress bar can be implemented through the R-package `progressr`, but for the most informative progress bar, this requires that this function is implemented within the mice sampler. I think this should be doable.

**Reply by pehkawn** [reply]
Thanks for the tip! I will look into `progressr` for now, and hope to see this feature added in `mice()` or `futuremice()` soon.
